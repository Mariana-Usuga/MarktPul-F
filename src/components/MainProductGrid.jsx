import React from 'react';
import ProductItem from './ProductItem';
import '../styles/components/MainProductGrid.scss';

const MainProductGrid = () => {
  const data = [
    {
      id: 1,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 2,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 3,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 4,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 5,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 6,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 7,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 8,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 9,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 10,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 11,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 12,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 13,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 14,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 15,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 16,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 17,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 18,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 19,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 20,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 21,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 22,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 23,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 24,
      tipoMercado: 'Fisico',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG1', 'TAG2', 'TAG2'],
      img: 'https://via.placeholder.com/220x172',
    },
    {
      id: 25,
      tipoMercado: 'Digital',
      infoMercado: 'Loremipsumm',
      tagMercado: ['TAG3', 'TAG4', 'TAG5'],
      img: 'https://via.placeholder.com/220x172',
    },
  ];
  return (
    <div className="container">
      {data.map((product) => {
        const { tipoMercado, infoMercado, tagMercado, img, id } = product;
        return (
          <ProductItem
            key={id}
            tipoMercado={tipoMercado}
            infoMercado={infoMercado}
            tagMercado={tagMercado}
            img={img}
          />
        );
      })}
    </div>
  );
};
export default MainProductGrid;
