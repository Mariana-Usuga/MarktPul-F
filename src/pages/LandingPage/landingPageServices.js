import axios from 'axios';

const URL_BASE = process.env.REACT_APP_API_URL_BASE;

export const getMarkets = () => axios.get(`${URL_BASE}/api/market`);
export const getProducts = () => axios.get(`${URL_BASE}/api/product`);
