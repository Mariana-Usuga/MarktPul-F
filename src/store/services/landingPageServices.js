import axios from 'axios';

const URL_BASE = process.env.REACT_APP_API_URL_BASE || 'http://localhost:8080';

export const getMarkets = () => axios.get(`${URL_BASE}/api/market`);
export const getProducts = () => axios.get(`${URL_BASE}/api/product`);

export const getUser = (token) => {
  const url = `${URL_BASE}/api/user/me`;
  const config = {
    method: 'get',
    url,
    headers: {
      Authorization: `Bearer ${token}`,
      'Content-Type': 'application/json',
    },
  };
  return axios(config);
};
