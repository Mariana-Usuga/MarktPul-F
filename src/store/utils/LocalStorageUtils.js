const getCurrentLocalStorage = (item) => {
  try {
    const localItem = JSON.parse(localStorage.getItem(`${item}`)) ?? '';
    return localItem;
  } catch {
    return '';
  }
};

export { getCurrentLocalStorage };
