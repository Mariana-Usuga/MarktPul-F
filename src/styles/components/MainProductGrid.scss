
.container{
    width: 100%;
    display: flex;
    gap: 20px;
    flex-wrap: wrap;
    //justify-content: space-around;
    justify-content: center;
}

@media(min-width:992px){
    .container{
        width:80%;
    }
}
