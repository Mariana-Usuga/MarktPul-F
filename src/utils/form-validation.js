function validateEmail(email) {
  return (
    /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/.test(email) && email.length > 0
  );
}

function validateUsername(uname) {
  return uname.length > 4;
}

function validateSubmit(errors) {
  const isValid = Object.keys(errors).filter(
    (key) => errors[key].length > 0,
  ).length;
  return isValid;
}

function validatePassword(password) {
  return (
    /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/.test(password)
    && password.length > 0
  );
}

export {
  validateEmail, validatePassword, validateSubmit, validateUsername,
};
