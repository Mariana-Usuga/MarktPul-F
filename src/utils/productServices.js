import axios from 'axios';

// const URL_BASE = process.env.REACT_APP_API_URL_BASE;

export const postProduct = async (formProduct) => {
  const reponseProduct = await axios.post(
    'http://localhost:3002/api/product',
    formProduct,
  );

  return reponseProduct.data;
};
