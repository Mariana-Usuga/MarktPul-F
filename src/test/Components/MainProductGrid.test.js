import React from 'react';
// import jest from '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import MainProductGrid from '../../components/MainProductGrid';
import { getMarkets } from '../../utils/landingPageServices';

describe('Pruebas en Main ProductGrid', () => {
  test('debe mostrar el componente correctamente ', () => {
    const wrapper = shallow(<MainProductGrid />);
    expect(wrapper).toMatchSnapshot();
  });
  test('debe traer 6 elementos', async () => {
    const testGetMarkets = await getMarkets();
    expect(testGetMarkets.data.length).toBe(6);
  });
  /* test('debe mostrar los markets cuando se cargan', () => {
    const testMarkets = [
      {
        _id: '61d9a7393b950197f5fde04c',
        title: 'Mercado de Mariana Usuga',
        description: 'Vende tus productos en el mejor mercado de la ciudad',
        organizer: 'Mariana Usuga Montoya',
        image:
          'https://res.cloudinary.com/db3njhxi0/image/upload/v1639700587/Markt-Pul/Markets/mercadoPorDefecto_aukqrf.jpg',
        __v: 0,
        place: 'Cr 59 # 235-61',
      },
    ];
    getMarkets.mockReturnValue({
      data: testMarkets,
    });
    const wrapper = shallow(<MainProductGrid />);
    expect(wrapper).toMatchSnapshot();
  }); */
});
