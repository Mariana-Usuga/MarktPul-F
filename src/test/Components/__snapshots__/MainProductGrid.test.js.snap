// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Main ProductGrid debe mostrar el componente correctamente  1`] = `
<div
  className="container"
/>
`;
